### ISSUE TITLE
Unit tests to noisy

### ISSUE THREAD

**Issue by dschwoerer** [boutproject/BOUT-dev#1039]
The unit tests produce roughly 1200 lines.

If one fails, I am having troubles finding that one.

Without the unit tests there would be only ~500 lines on travis.

**Comment by d7919** [comment]
I think the failed tests are summarised at the end of the unit test run so I think the easiest way at the moment is to just scroll straight to the bottom of the log and then skip up a bit.

I think it's possible to set it up such that it's much quieter @ZedThree can probably comment on if it's possible to do this whilst still retaining the rest of the useful reporting.

**Comment by dschwoerer** [comment]
I think the output of the failed test is not printed, just that it failed.

I couldn't find a way to make it less verbose, except to disable all output.
There is a way to write to xml, which could be parsed and summarized - but I didn't want to do that unless there is no better way ...

**Comment by ZedThree** [comment]
It's pretty noisy out of the box, but we could write our own event listener (see https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md)

Or use something like: https://github.com/joshkel/gtpp

I'd rather not require gtpp just to run tests, but I'm happy to use it on Travis. The event listener might be a better way to go, though it would mean maintaining it.

**Comment by d7919** [comment]
One other option for Travis would be to capture the output of running the unit tests and then parse this for the tests that failed and then rerun just those using the `--gtest_filter` runtime option.

Alternatively piping through something like `| awk '/\[ RUN      \]/{skip=1;next}/\[       OK ]/{skip=1;next}/\[  FAILED  \]/{print;skip=0}skip' ` would strip out the tests that pass, leaving just the failure cases and reports for each test family (with timing). The test family output can also be removed by adding `/\[----------\]/{skip=1;next}` to the patterns to match.

**Comment by dschwoerer** [comment]
I like the awk version, so I extended it:

```
$ ./serial_tests | ./quiet.sh 
Running main() from bout_test_main.cxx
[==========] Running 587 tests from 31 test cases.
[----------] Global test environment set-up.
[----------] 1 test from BoutMeshTest
[----------] 1 test from BoutMeshTest (0 ms total)
[ RUN      ] MeshTest.Ind3DTo2D
mesh/test_mesh.cxx:88: Failure
      Expected: localmesh.ind3Dto2D(index3d_49)
      Which is: 7
To be equal to: Ind2D(3)
      Which is: 3
[  FAILED  ] MeshTest.Ind3DTo2D (1 ms)
[----------] 10 tests from MeshTest (1 ms total)
[----------] 4 tests from AssertTest (0 ms total)
[----------] 8 tests from ArrayTest (0 ms total)
[----------] 26 tests from RegionTest (0 ms total)
[----------] 21 tests from RegionIndexTest/0 (0 ms total)
[----------] 21 tests from RegionIndexTest/1 (0 ms total)
[----------] 23 tests from FieldIndexTest/0 (0 ms total)
[----------] 23 tests from FieldIndexTest/1 (1 ms total)
[----------] 26 tests from FieldGroupTest (0 ms total)
[----------] 68 tests from Field2DTest (3 ms total)
[----------] 107 tests from Field3DTest (8 ms total)
[----------] 84 tests from FieldPerpTest (2 ms total)
[----------] 1 test from Vector2DTest (1 ms total)
[----------] 1 test from Vector3DTest (0 ms total)
[----------] 20 tests from OutputTest (0 ms total)
[----------] 14 tests from OptionsReaderTest (1 ms total)
[----------] 5 tests from MsgStackTest (0 ms total)
[ RUN      ] OptionsTest.GetBoolFromString
        Option :bool_key3 = A_bool_starts_with_T_or_N_or_Y_or_F_or_1_or_0 (code) overwritten with:
                :bool_key3 = yes_this_is_a_bool (code2)
[----------] 27 tests from OptionsTest (0 ms total)
[----------] 2 tests from BoutExceptionTest (734 ms total)
[----------] 1 test from BoutRhsFailTest (0 ms total)
[----------] 2 tests from BoutParallelRhsFailTest (0 ms total)
[----------] 1 test from BoutIterationFailTest (0 ms total)
[----------] 1 test from OldMatrixTest (0 ms total)
[----------] 11 tests from MatrixTest (0 ms total)
[----------] 11 tests from TensorTest (0 ms total)
[----------] 4 tests from Invert3x3Test (1 ms total)
[----------] 13 tests from NumberUtilitiesTest (0 ms total)
[----------] 15 tests from StringUtilitiesTest (0 ms total)
[----------] 11 tests from ExpressionParserTest (1 ms total)
[----------] 25 tests from RangeTest (0 ms total)
[==========] 587 tests from 31 test cases ran. (753 ms total)
[  PASSED  ] 586 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] MeshTest.Ind3DTo2D
 1 FAILED TEST
```

And the script is:
```
$ cat quiet.sh 
#!/usr/bin/env sh
  awk '{
    if (match($0,"^$")) {
        empty=1
        #print "empty", $0
        if (running)
            out=1
        else
            out=0
        
    } else {
        if (match($0,"\\[ *RUN *\\]")) {
            #print "run", $0
            running=$0;
            out=0
        } else if (match($0,"\\[ *OK *\\]")) {
            #print "ok", $0
            running=""
            out=0
        } else if (match($0,"\\[ *FAILED *\\]")) {
            #print "failed", $0
            if (running)
                print running
            out=1
        } else if (match($0,"\\[-*\\]")) {
            #print "over", empty,"x",$0
            if (empty)
                out=0
            else
                out=1
        } else {
            if (running){
                print running
                running=""
            }
            out=1
        }
        empty=0
    }
    if (out)
        print $0
}'
```

**Comment by ZedThree** [comment]
Fixed with #1052
